backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/TopDownAnalyzerFacadeForKonan.kt

```diff
@@ -80,3 +97,32 @@ object TopDownAnalyzerFacadeForKonan {
         }
     }
 }
+
+private class ResolvedDependencies(
+        resolvedLibraries: KonanLibraryResolveResult,
+        storageManager: StorageManager,
+        builtIns: KotlinBuiltIns,
+        specifics: LanguageVersionSettings,
+        friendLibsSet: Set<File>?
+) {
+
+    val moduleDescriptors: KonanResolvedModuleDescriptors
+    val friends: Set<ModuleDescriptorImpl>
+
+    init {
+
+        val collectedFriends = mutableListOf<ModuleDescriptorImpl>()
+
+        val customAction: (KonanLibrary, ModuleDescriptorImpl) -> Unit = { library, moduleDescriptor ->
+            friendLibsSet?.apply {
```

Comment:
It is possible to write this code like
```
if (friendLibsSet?.contains(library.libraryFile) == true) {
    collectedFriends.add(moduleDescriptor)
}
```
Also what is the reason to have `friendLibsSet == null` instead of using `emptySet()`?

Reply:
> It is possible to write this code like

Will fix.

> Also what is the reason to have friendLibsSet == null instead of using emptySet()?

This line of code was written by someone else. I just moved it to the new place. But I agree with you. This sounds reasonable. Will fix.